src/parse.c

```diff
@@ -1105,12 +1105,76 @@ handle_addresses(yaml_document_t* doc, yaml_node_t* node, const void* _, GError*
     return handle_generic_addresses(doc, node, TRUE, &(cur_netdef->ip4_addresses), &(cur_netdef->ip6_addresses), error);
 }
 
+const static char *GW_MULTIPLE_IP_ERROR = "gateway: there must be at most one of each IP type";
```

Comment:
I think we can move this into `handle_single_gateway()`, to keep it all together. It doesn't need to be a global variable, does it?

Also, please replace "IP type" => "IP address family" here as well.